diff -Nru gail.orig/gail.c gail/gail.c
--- gail.orig/gail.c	2011-01-31 11:24:51 +0900
+++ gail/gail.c	2011-03-02 20:48:46 +0900
@@ -307,9 +307,11 @@
    * then ignore the focus notification as the embedded
    * plug will report a focus notification.
    */
+#ifdef GDK_WINDOWING_X11
   if (GTK_IS_SOCKET (widget) &&
       gtk_socket_get_plug_window (GTK_SOCKET (widget)) != NULL)
     return TRUE;
+#endif
   /*
    * The widget may not yet be visible on the screen so we wait until it is.
    */
diff -Nru gail.orig/gailtoplevel.c gail/gailtoplevel.c
--- gail.orig/gailtoplevel.c	2011-01-31 11:24:51 +0900
+++ gail/gailtoplevel.c	2011-03-02 20:50:30 +0900
@@ -94,8 +94,11 @@
       if (!window || 
           !gtk_widget_get_visible (widget) ||
           is_attached_menu_window (widget) ||
-          gtk_widget_get_parent (GTK_WIDGET (window)) ||
-          GTK_IS_PLUG (window))
+          gtk_widget_get_parent (GTK_WIDGET (window))
+#ifdef GDK_WINDOWING_X11
+          || GTK_IS_PLUG (window)
+#endif
+          )
         {
           GList *temp_l  = l->next;
 
@@ -215,8 +218,11 @@
   widget = GTK_WIDGET (object);
   if (gtk_widget_get_parent (widget) ||
       is_attached_menu_window (widget) ||
-      is_combo_window (widget) ||
-      GTK_IS_PLUG (widget))
+      is_combo_window (widget)
+#ifdef GDK_WINDOWING_X11
+      || GTK_IS_PLUG (widget)
+#endif
+      )
     return TRUE;
 
   child = gtk_widget_get_accessible (widget);
diff -Nru gail.orig/gailwindow.c gail/gailwindow.c
--- gail.orig/gailwindow.c	2011-01-31 11:24:51 +0900
+++ gail/gailwindow.c	2011-03-02 20:51:20 +0900
@@ -226,8 +226,10 @@
 
       if (!g_strcmp0 (name, "gtk-tooltip"))
         obj->role = ATK_ROLE_TOOL_TIP;
+#ifdef GDK_WINDOWING_X11
       else if (GTK_IS_PLUG (widget))
         obj->role = ATK_ROLE_PANEL;
+#endif
       else if (gtk_window_get_window_type (GTK_WINDOW (widget)) == GTK_WINDOW_POPUP)
         obj->role = ATK_ROLE_WINDOW;
       else
